src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private final List<MachineProducer> machineProducers = List.of(new BulldozerProducer(),
+            new ExcavatorProducer(),
+            new TruckProducer());
+
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        for (MachineProducer machineProducer : machineProducers) {
+            if (machineProducer.getClassName().equals(type.toString())) {
```

Comment:
Better to use keyword `instance of` instead of comparing strings

Reply:
when I try do it **machineProducer.getClassName() instanceof type**  i have compile error: **Unknown class: 'type'**
when i do **type instanceof machineProducer.getClassName()** i have **Cannot resolve symbol 'getClassName'**

i will do machineProducer.getClassName() == type

if you can idea how i can realize it, hell me I'm haven't no idea i try everything . Thanks